src/dswx_sar/region_growing.py

```diff
@@ -0,0 +1,308 @@
+                    projection=meta_dict['projection'])
+
+
+def run(cfg):
+    '''
+    Run region growing with parameters in cfg dictionary
+    '''
+    logger.info('Starting DSWx-S1 Region-Growing')
+
+    t_all = time.time()
+
+    processing_cfg = cfg.groups.processing
+    outputdir = cfg.groups.product_path_group.scratch_path
+    pol_list = processing_cfg.polarizations
+    pol_str = '_'.join(pol_list)
+
+    # Region growing cfg
+    region_growing_cfg = processing_cfg.region_growing
+    region_growing_seed = region_growing_cfg.seed
+    region_growing_tolerance = region_growing_cfg.tolerance
+    region_growing_line_per_block = region_growing_cfg.line_per_block
+
+    fuzzy_tif_path = os.path.join(outputdir,
+                                 f'fuzzy_image_{pol_str}.tif')
+    water_meta = dswx_sar_util.get_meta_from_tif(fuzzy_tif_path)
+    water_tif_path = os.path.join(outputdir,
+                                 f"region_growing_output_binary_{pol_str}.tif")
+    temp_rg_tif_path = os.path.join(outputdir,
+                                   f'temp_region_growing_{pol_str}.tif')
+
+    # First, run region-growing algorithm for blocks
+    # to avoid to repeatly run with large image.
+    run_parallel_region_growing(fuzzy_tif_path,
+                                temp_rg_tif_path,
+                                lines_per_block=region_growing_line_per_block,
+                                initial_seed=region_growing_seed,
+                                tolerance=region_growing_tolerance,
+                                maxiter=0)
+
+    fuzzy_map = dswx_sar_util.read_geotiff(fuzzy_tif_path)
+    temp_rg = dswx_sar_util.read_geotiff(temp_rg_tif_path)
+
+    # replace the fuzzy values with 1 for the pixels
+    # where the region-growing already applied
+    fuzzy_map[temp_rg == 1] = 1
+
+    # Run region-growing again for entire image
+    region_grow_map = region_growing(
+                            fuzzy_map,
+                            initial_seed=region_growing_seed,
+                            tolerance=region_growing_tolerance,
+                            maxiter=0)
+
+    dswx_sar_util.save_dswx_product(
+                region_grow_map,
+                water_tif_path,
+                geotransform=water_meta['geotransform'],
+                projection=water_meta['projection'],
+                description='Water classification (WTR)',
+                scratch_dir=outputdir)
+
+    t_all_elapsed = time.time() - t_all
+    logger.info(f"successfully ran region growing in {t_all_elapsed:.3f} seconds")
+
+
+def main():
+    'Main Function to run region growing'
+    parser = _get_parser()
+
+    args = parser.parse_args()
+
+    mimetypes.add_type("text/yaml", ".yaml", strict=True)
+    flag_first_file_is_text = 'text' in mimetypes.guess_type(
+        args.input_yaml[0])[0]
+
+    if flag_first_file_is_text:
+        cfg = RunConfig.load_from_yaml(args.input_yaml[0], 'dswx_s1', args)
+    run(cfg)
```

Comment:
```suggestion
    if 'text' not in mimetypes.guess_type(
        args.input_yaml[0])[0]:
        raise ValueError('input yaml file is not text')

    cfg = RunConfig.load_from_yaml(args.input_yaml[0], 'dswx_s1', args)
    run(cfg)
```
Add exception raising if input not recognized. Current implementation will pass undefined/uninitialized `cfg` otherwise.

Reply:
Thank you for good suggestions :)